core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/validation/service/PythonVersionValidator.java

```diff
@@ -0,0 +1,41 @@
+package com.sequenceiq.cloudbreak.service.upgrade.validation.service;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
+import com.sequenceiq.cloudbreak.common.exception.UpgradeValidationFailedException;
+import com.sequenceiq.cloudbreak.service.upgrade.UpgradeImageInfo;
+import com.sequenceiq.cloudbreak.service.upgrade.validation.PythonVersionBasedRuntimeVersionValidator;
+
+@Component
+public class PythonVersionValidator implements ServiceUpgradeValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PythonVersionValidator.class);
+
+    @Inject
+    private PythonVersionBasedRuntimeVersionValidator pythonVersionBasedRuntimeVersionValidator;
+
+    @Override
+    public void validate(ServiceUpgradeValidationRequest validationRequest) {
+        UpgradeImageInfo upgradeImageInfo = validationRequest.getUpgradeImageInfo();
+        Image currentImage = upgradeImageInfo.getCurrentStatedImage().getImage();
+        Image targetImage = upgradeImageInfo.getTargetStatedImage().getImage();
+        if (isUpgradePermittedForRuntime(currentImage, targetImage)) {
```

Comment:
A rename of the `isUpgradePermittedForRuntime` method would be great here to `isUpgradeNotPermittedForRuntime` or `isUpgradeDeniedForRuntime`

Reply:
cc: @tiborpopovics